### ISSUE TITLE
Logitech MX Master Gesture button on Fedora for continuous volume slider

### ISSUE THREAD

**Issue by gardellimarius** [pwr-Solaar/Solaar#2211]
I'm trying to create a custom rule ( the third subrule ) where if I press the gesture button and move the mouse down the volume decreases.

It works, but I want it to behave differently.

I have the same mouse setup on Windows 11 and If I hold down the gesture button and move the mouse down the volume decreases until I stop moving.

With the current rules, on Fedora I have to press the gesture button and move the mouse down every time to lower the volume.

Basically I want it to behave like it does on Windows or Mac: where you press a single time on the gesture button & the volume gets higher/lower based on how much you move your mouse, instead of clicking and dragging the mouse every time.

![Solaar setup](https://github.com/pwr-Solaar/Solaar/assets/69801528/bfb6be57-153c-44b5-a64a-b0ed523912b7)
[Video showcasing behavior on Fedora](https://imgur.com/a/HlwhyLU)

**Comment by pfps** [comment]
This would be a reasonable extension, but may take a while to implement.

But how would the result be communicated to something that would control the volume?

**Comment by OleMussmann** [comment]
I could imagine a "repeat" option for key presses. Is it internally known when a gesture starts and ends? If so, with the "repeat" option enabled, a repeated keypress could be sent until the gesture ends. This way the `XF86_AudioLowerVolume` could be triggered multiple times. This would be time-based, not distance-based though.

**Comment by pfps** [comment]
What program does the volume sliding in Windows?

**Comment by jurf** [comment]
Logitech Options has this kind of behaviour by default if that is what you are asking.

**Comment by pfps** [comment]
What does the user interface look like?   Is there a gesture that does this?  Can this gesture be changed?

**Comment by jurf** [comment]
Not really, some actions, like volume, zooming or panning, just fire per unit of distance travelled.

![obrázok](https://github.com/pwr-Solaar/Solaar/assets/7472425/1af6027d-9ec4-4de7-98d4-d4f8cded4e60)

I also noticed that I do not need to let go of the gesture buttons to trigger a gesture in Options+, I quite like that.

I could picture this being implemented with a simple "fire continuosly" checkbox (and maybe a checkbox to fire asap, before lifting the gesture button?).

**Comment by pfps** [comment]
It looks like Solaar gestures work differently, as all the gesture movements and key events are collected and then sent to the rule system.  I don't see a way to reconcile these two.

As far as continuous volume slider is concerned, it might be possible to report on the total distance and use that in the rule.   But then what would the rule action do with this?

**Comment by teddy8193** [comment]
I wonder if this there is any update on it.
I'd also like to control the browser zooming, which can be done with Ctrl + Scroll in keyboard, like the sequence with the mouse would be, Press and hold gesture button -> Mouse up gesture -> Continuously executing Ctrl+ScrollUp to zoom in -> Release the gesture button.

**Comment by jurf** [comment]
> Closing due to no response.

I am not sure what response did you want? Total distance would not be useful for my usecase. It's a shame if this cannot be implemented (although I am not exactly sure why these modes conflict), it's one of the coolest features of the gesture button.

**Comment by pfps** [comment]
The problem is having the rule system respond continually during movement versus it responding after the gesture button is released.  Solaar is set up to do the latter.  Having rules run for every mouse movement requires a different interface that is not planned.   If someone wants to do the programming then go ahead.

**Comment by RedyAu** [comment]
Hello all on this thread, I've tried implementing this feature in PR #2979.
[Here's a video preview of what's achieved.](https://www.youtube.com/watch?v=EZTrBE6s3Tw) Would this solve your usecases?
